assets/js/settings.js

```diff
@@ -172,6 +172,43 @@ jQuery( document ).ready( function ( $ ) {
 		fetch( ajaxurl, { method: 'POST', body: data } );
 	}
 
+	/***** Unique Page Settings *****/
+
+	$( '.sensei-settings-unique-page select' ).on(
+		'change',
+		disableSelectedUniquePageOptions
+	);
+
+	/**
+	 * Disable page options that are already selected in another setting.
+	 */
+	function disableSelectedUniquePageOptions() {
+		const selectedPageIds = [];
+		const $inputs = $( '.sensei-settings-unique-page select' );
+
+		$inputs.each( function () {
+			selectedPageIds.push( $( this ).val() );
+		} );
+
+		$inputs.each( function () {
+			const currentValue = $( this ).val();
+
+			$( this )
+				.find( 'option' )
+				.prop( 'disabled', false )
+				.filter( function () {
+					const optionValue = $( this ).val();
+					return (
+						optionValue !== currentValue &&
+						selectedPageIds.includes( optionValue )
+					);
+				} )
+				.prop( 'disabled', true );
+		} );
+	}
```

Comment:
I think we should leave the "Select a page" option in this logic, otherwise, It considers "Select a Page" as a unique option too and it gets disabled for other pages if selected for one of the pages-

https://github.com/Automattic/sensei/assets/6820724/e6fb5291-9eb7-4cee-89a1-8964b173e581